projects/admin/src/app/circulation/patron/pending/pending-item/pending-item.component.ts

```diff
@@ -58,4 +75,40 @@ export class PendingItemComponent implements OnInit {
       );
     }
   }
+
+  canCancelRequest(): boolean {
+    // No permission API in backend
+    // Allowed when loan state is PENDING or ITEM_IN_TRANSIT_FOR_PICKUP according to actions.md
+    const itemStatus = [LoanState.PENDING, LoanState.ITEM_IN_TRANSIT_FOR_PICKUP, LoanState.ITEM_AT_DESK];
+    return itemStatus.some((element) => element === this.loan.metadata.state);
+  }
+
+  /** Show a confirmation dialog box for cancel request. */
+  showCancelRequestDialog(): void {
+    const config = {
+      ignoreBackdropClick: true,
+      initialState: {
+        title: this._translateService.instant('Cancel request'),
+        body: this._translateService.instant('Do you really want to delete this request?'),
+        confirmButton: true,
+        cancelTitleButton: this._translateService.instant('No'),
+        confirmTitleButton: this._translateService.instant('Yes')
+      }
+    };
+    this._dialogService.show(config).subscribe((confirm: boolean) => {
+      if (confirm) {
+        this._loanService.cancelLoan(
+          this.loan.metadata.item.pid,
+          this.loan.metadata.pid,
+          this._userService.user.currentLibrary
+        ).subscribe(() => {
+          this._toastrService.warning(
+            this._translateService.instant('The pending request has been cancelled.'),
+            this._translateService.instant('Request')
+          );
+          this.cancelRequestEvent.emit(this.loan.id);
+        });
+      }
+    });
```

Comment:
It could be of interest to extract this into a separate function that could be called (in future) to cancel multiple loans.